File: app/src/main/java/com/cleanup/todoc/domaine/usecase/GetProjectsUseCase.java
package com.cleanup.todoc.domaine.usecase;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.Transformations;

import com.cleanup.todoc.data.repository.ProjectRepository;
import com.cleanup.todoc.domaine.ProjectDomainUiMapper;
import com.cleanup.todoc.domaine.model.ProjectDomain;
import com.cleanup.todoc.presentation.model.Project;

import java.util.List;

public class GetProjectsUseCase {
    private final ProjectRepository repository;

    public GetProjectsUseCase(ProjectRepository repository) {
        this.repository = repository;
    }

    public LiveData<List<Project>> getAllProjects() {
        return Transformations.map(repository.getAllProjects(), ProjectDomainUiMapper::mapToProjectList);

    }
}

File: app/src/main/java/com/cleanup/todoc/domaine/usecase/GetProjectsUseCase.kt
package com.cleanup.todoc.domaine.usecase

import com.cleanup.todoc.data.repository.ProjectRepository
import com.cleanup.todoc.domaine.ProjectDomainUiMapper
import com.cleanup.todoc.domaine.model.ProjectDomain
import com.cleanup.todoc.presentation.model.Project
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map

class GetProjectsUseCase(private val repository: ProjectRepository) {
    val allProjects: Flow<List<Project>>
        get() = repository.allProjects.map { projects: List<ProjectDomain> ->
            ProjectDomainUiMapper.mapToProjectList(
                projects
            )
        }
}
